js/tests/taxonomyAssessor.test.js

```diff
@@ -0,0 +1,172 @@
+let Assessor = require( "../src/assessors/taxonomyAssessor.js" );
+let Paper = require("yoastseo/js/values/Paper.js");
+let factory = require( "./helpers/factory.js" );
+let i18n = factory.buildJed();
+let assessor = new Assessor( i18n );
+
+let getResults = function ( Results ) {
+	let assessments = [];
+
+	for ( let Result of Results ) {
+		assessments.push( Result._identifier );
+	}
+
+	return assessments;
+};
+
+describe ( "running assessments in the assessor", function() {
+	it( "runs assessments without any specific requirements", function() {
+		assessor.assess( new Paper( "" ) );
+		let AssessmentResults = assessor.getValidResults();
+		let assessments = getResults( AssessmentResults );
+
+		expect( assessments ).toEqual( [
+			"keyphraseLength",
+			"metaDescriptionLength",
+			"titleWidth",
+			"textLength"
+		] );
+	} );
+
+	it( "additionally runs assessments that only require a text", function() {
+		assessor.assess( new Paper( "text" ) );
+		let AssessmentResults = assessor.getValidResults();
+		let assessments = getResults( AssessmentResults );
+
+		expect( assessments ).toEqual( [
+			"keyphraseLength",
+			"metaDescriptionLength",
+			"titleWidth",
+			"textLength"
+		] );
+	} );
+
+	it( "additionally runs assessments that require a keyword", function() {
+		assessor.assess( new Paper( "text", { keyword: "keyword" } ) );
+		let AssessmentResults = assessor.getValidResults();
+		let assessments = getResults( AssessmentResults );
+
+		expect( assessments ).toEqual( [
+			"introductionKeyword",
+			"metaDescriptionLength",
+			"titleKeyword",
+			"titleWidth",
+			"textLength"
+		] );
+	} );
+
+	it( "additionally runs assessments that require text and a keyword", function() {
```

Comment:
I'm not sure I fully know how these tests work, but this test seems to me the same as the previous one starting at line 44?

Reply:
You're right, that's a duplicate. I'll remove it!